lib/core/src/connection_pool.cpp

```diff
@@ -237,11 +281,39 @@ namespace irods
         }
 
         try {
-            // NOLINTNEXTLINE(bugprone-unused-raii)
-            query<RcComm>{ctx.conn.get(), "select ZONE_NAME where ZONE_TYPE = 'local'"};
+            // Check if the number of requests executed via this connection has been reached.
+            if (options_.refresh_connections_after_n_retrievals) {
+                if (ctx.retrieval_count >= *options_.refresh_connections_after_n_retrievals) {
+                    return false;
+                }
+            }
+
+            // Check age of connection.
             if (std::time(nullptr) - ctx.creation_time > refresh_time_) {
                 return false;
             }
+
+            // NOLINTNEXTLINE(bugprone-unused-raii)
+            query<RcComm>{ctx.conn.get(), "select ZONE_NAME where ZONE_TYPE = 'local'"};
```

Comment:
Can we leave an explanatory comment here? Not sure what this means.

Reply:
Will do.

This is verifying the connection is still usable.

Reply:
Done.